packages/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -4238,127 +4236,108 @@ export default class Meeting extends StatelessWebexPlugin {
+        // eslint-disable-next-line func-names
+        // eslint-disable-next-line prefer-arrow-callback
+        if (this.type === _CALL_) {
+          resolve();
+        }
+        const joiningTimer = setInterval(() => {
+          timerCount += 1;
+          if (this.meetingState === FULL_STATE.ACTIVE) {
+            clearInterval(joiningTimer);
             resolve();
           }
-          const joiningTimer = setInterval(() => {
-            timerCount += 1;
-            if (this.meetingState === FULL_STATE.ACTIVE) {
-              clearInterval(joiningTimer);
-              resolve();
-            }
 
-            if (timerCount === 4) {
-              clearInterval(joiningTimer);
-              reject(new Error('Meeting is still not active '));
-            }
-          }, 1000);
+          if (timerCount === 4) {
+            clearInterval(joiningTimer);
+            reject(new Error('Meeting is still not active '));
+          }
+        }, 1000);
+      }))
+      .then(() =>
+        logRequest(this.roap
+          .sendRoapMediaRequest({
+            sdp: this.mediaProperties.peerConnection.sdp,
+            roapSeq: this.roapSeq,
+            meeting: this // or can pass meeting ID
+          }), {
+          header: `${LOG_HEADER} Send Roap Media Request.`,
+          success: `${LOG_HEADER} Successfully send roap media request`,
+          failure: `${LOG_HEADER} Error joining the call on send roap media request, `
         }))
-        .then(() =>
-          logRequest(this.roap
-            .sendRoapMediaRequest({
-              sdp: this.mediaProperties.peerConnection.sdp,
-              roapSeq: this.roapSeq,
-              meeting: this // or can pass meeting ID
-            }), {
-            header: `${LOG_HEADER} Send Roap Media Request.`,
-            success: `${LOG_HEADER} Successfully send roap media request`,
-            failure: `${LOG_HEADER} Error joining the call on send roap media request, `
-          }))
-        .then(() => {
-          const {peerConnection} = this.mediaProperties;
-
-          return new Promise((resolve, reject) => {
-            if (peerConnection.connectionState === CONNECTION_STATE.CONNECTED) {
-              LoggerProxy.logger.info(`${LOG_HEADER} PeerConnection CONNECTED`);
-
-              resolve(peerConnection);
-
-              return;
-            }
-            // Check if Peer Connection is STABLE (connected)
-            const stabilityTimeout = setTimeout(() => {
-              if (peerConnection.connectionState !== CONNECTION_STATE.CONNECTED) {
-                // TODO: Fix this after the error code pr goes in
-                reject(createMeetingsError(30202, 'Meeting connection failed'));
-              }
-              else {
-                LoggerProxy.logger.info(`${LOG_HEADER} PeerConnection CONNECTED`);
-                resolve(peerConnection);
-              }
-            }, PC_BAIL_TIMEOUT);
-
-            this.once(EVENT_TRIGGERS.MEDIA_READY, () => {
```

Comment:
this existing code on master was wrong, because EVENT_TRIGGERS.MEDIA_READY is triggered from 2 places: when we start receiving remote media and when we set the local tracks before any media connection is even created, so this block of code has always been resolving too early. Now I had to fix it, otherwise the call to getStats() would not work as it was being called too early.